crawl-ref/source/player-equip.cc

```diff
@@ -196,6 +195,9 @@ void unequip_effect(equipment_type slot, int item_slot, bool meld, bool msg)
         _unequip_armour_effect(item, meld, slot);
     else if (slot >= EQ_FIRST_JEWELLERY && slot <= EQ_LAST_JEWELLERY)
         _unequip_jewellery_effect(item, msg, meld, slot);
+
+    if (item.cursed())
+        destroy_item(item);
```

Comment:
1. Did you really mean for this to destroy cursed items whenever you change forms?
2. It feels like this should be in the same place as *Fragile, wherever that is.

Reply:
*Fragile is in a complicated place because of message order issues; because the Ash message happens before any unwield/remove effects it always occurs in the correct place.

Good catch on the melding though!